app/controllers/UserController.py

```diff
@@ -0,0 +1,196 @@
+                'name': user.name,
+                'email': user.email,
+                'school': user.school,
+                'role': user.role,
+                'grade': user.grade,
+                'dob': user.dob,
+                'created_at': user.created_at,
+                'updated_at': user.updated_at
+            }
+            return jsonify(data)
+        else:
+            return jsonify({'error': 'User not found'}), 404
+        
+    except Exception as e:
+        print(e)
+
+def register():
+    try:
+        name = request.form.get('name')
+        email = request.form.get('email')
+        password = request.form.get('password')
+        school = request.form.get('school')
+        role = request.form.get('role')
+        grade = request.form.get('grade')
+        dob = 0
+
+        user = User(
+            name=name,
+            email=email,
+            school=school,
+            role = role,
+            grade=grade,
+            dob=dob
+            )
+        user.setPassword(password)
+        db.session.add(user)
+        db.session.commit()
+
+        return response.success('', 'User has been created')
+    except Exception as e:
+            print(e)
+
+def updateUser(id):
+    try:
+        name = request.form.get('name')
+        email = request.form.get('email')
+        password = request.form.get('password')
+        school = request.form.get('school')
+        role = request.form.get('role')
+        grade = request.form.get('grade')
+        updated_at = datetime.utcnow
+
+        data = [
+            {
+                'name': name,
+                'email': email,
+                'password': password,
+                'school': school,
+                'role': role,
+                'grade': grade,
+                'updated_at': updated_at
+            }
+        ]
+        
+        user = User.query.filter_by(id=id).first()
+
+        if user:
+            if name is not None:
+                user.name = name
+            if email is not None:
+                user.email = email
+            if password is not None:
+                user.password = password
+                user.setPassword(user.password)
+            if school is not None:
+                user.school = school
+            if role is not None:
+                user.role = role
+            if grade is not None:
+                user.grade = grade
+
+            db.session.commit()
+
+            return 'Data user has been updated'
+
+        else:
+            return 'Update data failed'
+    
+    except Exception as e:
+        print(e)
+
+def deleteUser(id):
+    try:
+        user = User.query.filter_by(id=id).first()
+        if not user:
+            return response.badRequest([], 'User not found')
+        db.session.delete(user)
+        db.session.commit()
+
+        return response.success('', 'Data user has been deleted')
+    
+    except Exception as e:
+        print(e)
+
+def userData(data):
```

Comment:
https://peps.python.org/pep-0008/#function-and-variable-names

Reply:
done